app/Http/Controllers/ViewAssetsController.php

```diff
@@ -201,82 +230,90 @@ public function getRequestedAssets()
-        }
+        $this->redirectOnError($findlog);
+        $logObj = array($logID);
+        $data = [
+            "ids" => urlencode(base64_encode(json_encode($logObj))),
+            "eulaList" => $findlog->item->getEula(),
+            "assetNames" => $findlog->item->present()->name()
+        ];
+
+        return view('account/accept-asset', $data);
     }
 
-    // Save the acceptance
-    public function postAcceptAsset(Request $request, $logID = null)
+    public function redirectOnError($log)
     {
-
         // Check if the asset exists
-        if (is_null($findlog = Actionlog::where('id', $logID)->first())) {
+        if (is_null($log)) {
             // Redirect to the asset management page
             return redirect()->to('account/view-assets')->with('error', trans('admin/hardware/message.does_not_exist'));
         }
-
-
-        if ($findlog->accepted_id!='') {
+        if (!empty($log->accepted_id)) {
             // Redirect to the asset management page
             return redirect()->to('account/view-assets')->with('error', trans('admin/users/message.error.asset_already_accepted'));
         }
-
         if (!Input::has('asset_acceptance')) {
             return redirect()->back()->with('error', trans('admin/users/message.error.accept_or_decline'));
         }
-
+        if (is_null($log->item)) {
+            return redirect()->to('account')->with('error', trans('admin/hardware/message.does_not_exist'));
+        }
+        if (!Company::isCurrentUserHasAccess($log->item)) {
+            return redirect()->route('requestable-assets')->with('error', trans('general.insufficient_permissions'));
+        }
         $user = Auth::user();
-
-        if (($findlog->item_type==Asset::class) && ($user->id != $findlog->item->assigned_to)) {
+        if (($log->item_type==Asset::class) && ($user->id != $log->item->assigned_to)) {
             return redirect()->to('account/view-assets')->with('error', trans('admin/users/message.error.incorrect_user_accepted'));
         }
+    }
 
-        if ($request->filled('signature_output')) {
-            $path = config('app.private_uploads').'/signatures';
-            $sig_filename = "siglog-".$findlog->id.'-'.date('Y-m-d-his').".png";
-            $data_uri = e($request->get('signature_output'));
-            $encoded_image = explode(",", $data_uri);
-            $decoded_image = base64_decode($encoded_image[1]);
-            file_put_contents($path."/".$sig_filename, $decoded_image);
-        }
-
-
-        $logaction = new Actionlog();
+    private function saveSignature($log, $signatureOutput)
+    {
+        $path = config('app.private_uploads').'/signatures';
+        $sig_filename = "siglog-".$log->id.'-'.date('Y-m-d-his').".png";
+        $data_uri = e($signatureOutput);
+        $encoded_image = explode(",", $data_uri);
+        $decoded_image = base64_decode($encoded_image[1]);
+        file_put_contents($path."/".$sig_filename, $decoded_image);
```

Comment:
I think this should be  using the new `Storage::` facades, so we can be compatible with users who want to store their files in S3. (I must have missed updating it there when I switched us over to flysystem)